src/sourcegraph.d.ts

```diff
@@ -717,6 +717,14 @@ declare module 'sourcegraph' {
         ): ProviderResult<Location[]>
     }
 
```

Comment:
I think you should just call it `QueryTransformer` not `QueryTransformerProvider`. I struggled with this in other cases but here's my thinking:

- The goal of naming is to communicate what something is.
- Using the "Provider" suffix is generally good for things like this because it communicates that this fits the "provider" pattern. Not in the strict Erich Gamma _Design Patterns_ sense, but just in the "this is like hover providers, definition providers, etc., in that it can be registered/unregistered and provides data given some parameters".
- But when there is a word that is similar to "provider", is more accurate, is short, and is the term you'd use anywhere else in code describing this, then you should use that word instead.
- In this case, that word is "Transformer". QueryTransformer and QueryTransformerProvider basically mean the same thing; in the latter, "transformer" and "provider" basically mean the same thing, so it's weird to have them both.
- Looking at [vscode.d.ts](https://sourcegraph.com/github.com/Microsoft/vscode/-/blob/src/vs/vscode.d.ts#), they have many things that follow the guidelines I laid out instead of always using the "Provider" suffix: registerUriHandler, registerWebviewPanelSerializer, etc. Those are way better than registerUriHandlerProvider or registerUriProvider, or registerWebviewPanelSerializationProvider or registerWebviewPanelSerializerProvider.